src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostDocumentSpellCheckEndpointTest.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
+
+public class CohostDocumentSpellCheckEndpointTest(ITestOutputHelper testOutputHelper) : CohostEndpointTestBase(testOutputHelper)
+{
+    [Fact]
+    public async Task Handle()
+    {
+        var input = """
+            @page [|"this is csharp"|]
+
+            <div>[|
+
+                Eat more chickin.
+
+            |]</div>
+
+            <script>
+                // no spell checking of script tags
+                @([|"unless they contain csharp"|])
+            </script>
+
+            <style>
+                // no spell checking of style tags
+                @([|"unless they contain csharp"|])
+            </style>
+
+            @{ var [|x|] = [|"csharp"|];
+
+            @*[| Eat more chickin. |]*@
+
+            <div class="[|fush|]" />
+
+            @code
+            {
+                void [|M|]()
+                {
+                    [|// Eat more chickin|]
+                }
+            }
+            """;
```

Comment:
Asking to make sure I understood, is this unit test only testing if the spell check endpoint is correctly finding the spell checkable ranges within a razor document?

Assuming that everything in between `[|` and `|]` is supposed to be all the spell checkable ranges that spell check endpoint finds. Based on this test, it should include both C# and html ranges

Reply:
Correct, our job is just to identify the ranges, not do the actual spell checking :)